trie/trie.go

```diff
@@ -415,13 +415,17 @@ func (t *Trie) delete(n node, prefix, key []byte) (bool, node, error) {
 		// value that is left in n or -2 if n contains at least two
 		// values.
 		pos := -1
-		for i, cld := range &n.Children {
-			if cld != nil {
-				if pos == -1 {
-					pos = i
-				} else {
-					pos = -2
-					break
+		if nn != nil {
+			pos = -2
+		} else {
```

Comment:
I find this hard to intuitively verify. I think it might be easier to follow if you break here, so replace this with 
```golang
if nn != nil{	
	// n still contains at least two values and cannot be reduced.
	return true, n, nil
}
```
As far as I can tell, that^ change should be equivalent with your change. However, looking at it, I still don't see why this is necessarily true/correct?